src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerCorporateValidator.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.ManagingOfficerCorporateDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.CountryLists;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.StringValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import java.util.List;
+
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationUtils.getQualifiedFieldName;
+
+@Component
+public class ManagingOfficerCorporateValidator {
+    
+    private final AddressDtoValidator addressDtoValidator;
+    
+    @Autowired
+    public ManagingOfficerCorporateValidator(AddressDtoValidator addressDtoValidator) {
+        this.addressDtoValidator = addressDtoValidator;
+    }
+    
+    public Errors validate(List<ManagingOfficerCorporateDto> managingOfficerCorporateDtoList, Errors errors, String loggingContext) {
+        for (ManagingOfficerCorporateDto managingOfficerCorporateDto : managingOfficerCorporateDtoList) {
+
+            validateName(managingOfficerCorporateDto.getName(), errors, loggingContext);
+            validateAddress(ManagingOfficerCorporateDto.PRINCIPAL_ADDRESS_FIELD, managingOfficerCorporateDto.getPrincipalAddress(), errors, loggingContext);
+
+            boolean sameAddressFlagValid = validateServiceAddressSameAsPrincipalAddress(managingOfficerCorporateDto.getServiceAddressSameAsPrincipalAddress(), errors, loggingContext);
+            if (sameAddressFlagValid && Boolean.FALSE.equals(managingOfficerCorporateDto.getServiceAddressSameAsPrincipalAddress())) {
+                validateAddress(ManagingOfficerCorporateDto.SERVICE_ADDRESS_FIELD, managingOfficerCorporateDto.getServiceAddress(), errors, loggingContext);
+            }
+
+            validateLegalForm(managingOfficerCorporateDto.getLegalForm(), errors, loggingContext);
+            validateLawGoverned(managingOfficerCorporateDto.getLawGoverned(), errors, loggingContext);
+
+            boolean onRegister = validateOnRegisterInCountryFormedIn(managingOfficerCorporateDto.getOnRegisterInCountryFormedIn(), errors, loggingContext);
+            if (onRegister && Boolean.TRUE.equals(managingOfficerCorporateDto.getOnRegisterInCountryFormedIn())) {
+                validatePublicRegisterName(managingOfficerCorporateDto.getPublicRegisterName(), errors, loggingContext);
+                validateRegistrationNumber(managingOfficerCorporateDto.getRegistrationNumber(), errors, loggingContext);
+            }
```

Comment:
Same here this is currently in review: https://github.com/companieshouse/overseas-entities-api/pull/107